### PR TITLE
make artifacts conditional

### PR DESCRIPTION
**Pull request by zikalino**
##### SUMMARY
DTL artifacts tests should depend on GitHub token being availanle.
In this way it will be tested in azure_preview_modules and disabled here.

##### ISSUE TYPE
- Bugfix Pull Request

##### COMPONENT NAME
azure_rm_devtestlabartifactsource

### CODE REVIEW

File: test/integration/targets/azure_rm_devtestlab/tasks/main.yml [hunk 1]
```diff
@@ -173,51 +173,69 @@
...
+    source_type: github
+    folder_path: /tasks
+    security_token: "{{ github_token }}"
+  register: output 
+  when: "github_token | length > 0"
+- name: Assert if the change was correctly reported
+  assert:
+    that:
+      - output.changed
+  when: "github_token | length > 0"
```
Comment (yungezz) [inline test/integration/targets/azure_rm_devtestlab/tasks/main.yml:191]: this is  always false, right? when it could be true?

Comment (zikalino) [reply test/integration/targets/azure_rm_devtestlab/tasks/main.yml:191]: in azure_preview_modules will be true
or if somebody adds GitHub_token in the env

Comment (zikalino) [reply test/integration/targets/azure_rm_devtestlab/tasks/main.yml:191]: I am testing the same change in azure_preview_modules as well.
the idea is that we have the same test in both.

**Review by yungezz — APPROVED** [review]

**Comment by yungezz** [comment]
shipit

**Pull request merged**